dm/master/server.go

```diff
@@ -684,6 +683,9 @@ func (s *Server) ShowDDLLocks(ctx context.Context, req *pb.ShowDDLLocksRequest)
 		resp.Locks = append(resp.Locks, l)
 	}
 
+	// show optimistic locks.
+	resp.Locks = append(resp.Locks, s.optimist.ShowLocks(req.Task, req.Sources)...)
```

Comment:
Should we use some field to differ optimist lock from pessimist ones? Since their definitions are different.

Reply:
how about adding a field named `mode`?

Reply:
I think it's OK

Reply:
addressed in https://github.com/pingcap/dm/pull/580/commits/a02a9cc269f8647a8cd9b6d30cae116eb1a34df8.